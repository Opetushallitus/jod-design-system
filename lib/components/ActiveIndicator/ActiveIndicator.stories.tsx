import type { Meta, StoryObj } from '@storybook/react';

import { ActiveIndicator } from './ActiveIndicator';

const meta = {
  title: 'ActiveIndicator',
  component: ActiveIndicator,
  tags: ['autodocs'],
} satisfies Meta<typeof ActiveIndicator>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <ul className="text-button-md">
      <li className="flex items-center">
        <ActiveIndicator /> Item 1
      </li>
      <li className="flex items-center">Item 2</li>
      <li className="flex items-center">Item 3</li>
    </ul>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Simple indicator for an active list item',
      },
    },
  },
};
