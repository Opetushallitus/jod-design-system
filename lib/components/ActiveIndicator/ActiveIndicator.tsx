export const ActiveIndicator = () => (
  <svg height="15" width="15" aria-hidden focusable="false" className="ds:absolute ds:-ml-6">
    <circle cx="7.5" cy="7.5" r="7.5" className="ds:fill-accent" />
  </svg>
);
