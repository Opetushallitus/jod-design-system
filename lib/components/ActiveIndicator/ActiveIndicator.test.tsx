import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { ActiveIndicator } from './ActiveIndicator';

describe('ActiveIndicator', () => {
  it('renders the ActiveIndicator component correctly', () => {
    const { container } = render(<ActiveIndicator />);
    const svgElement = container.querySelector('svg');
    expect(svgElement).toBeInTheDocument();
    expect(svgElement).toHaveAttribute('height', '15');
    expect(svgElement).toHaveAttribute('width', '15');
    expect(container.firstChild).toMatchSnapshot();
  });
});
