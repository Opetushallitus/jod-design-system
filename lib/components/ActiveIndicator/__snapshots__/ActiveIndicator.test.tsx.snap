// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActiveIndicator > renders the ActiveIndicator component correctly 1`] = `
<svg
  aria-hidden="true"
  class="ds:absolute ds:-ml-6"
  focusable="false"
  height="15"
  width="15"
>
  <circle
    class="ds:fill-accent"
    cx="7.5"
    cy="7.5"
    r="7.5"
  />
</svg>
`;
